operator/internal/manifests/openshift/recordngrule.go

```diff
@@ -5,23 +5,34 @@ import lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
 func RecordingRuleTenantLabels(r *lokiv1.RecordingRule) {
 	switch r.Spec.TenantID {
 	case tenantApplication:
-		for groupIdx, group := range r.Spec.Groups {
-			group := group
-			for ruleIdx, rule := range group.Rules {
-				rule := rule
-				if rule.Labels == nil {
-					rule.Labels = map[string]string{}
-				}
-				rule.Labels[opaDefaultLabelMatcher] = r.Namespace
-				group.Rules[ruleIdx] = rule
-			}
-			r.Spec.Groups[groupIdx] = group
-		}
-	case tenantInfrastructure, tenantAudit:
-		// Do nothing
-	case tenantNetwork:
-		// Do nothing
+		appendRecordingRuleLabels(r, map[string]string{
+			opaDefaultLabelMatcher:    r.Namespace,
+			ocpMonitoringGroupByLabel: r.Namespace,
+		})
+	case tenantInfrastructure, tenantAudit, tenantNetwork:
+		appendRecordingRuleLabels(r, map[string]string{
+			ocpMonitoringGroupByLabel: r.Namespace,
+		})
 	default:
 		// Do nothing
 	}
 }
+
+func appendRecordingRuleLabels(ar *lokiv1.RecordingRule, labels map[string]string) {
+	for groupIdx, group := range ar.Spec.Groups {
+		group := group
+		for ruleIdx, rule := range group.Rules {
+			rule := rule
```

Comment:
I've tested (using our unit test for `RecordingRuleTenantLabels`) and we get the same result so I'm curious if I'm missing anything or if we can just remove these two
```suggestion
		for ruleIdx, rule := range group.Rules {
```